packages/babel-plugin-redux-saga/test/fixtures/preset-env/source.js

```diff
@@ -5,3 +5,13 @@ function* test1() {
 function* test2() {
   yield 2
 }
+
+class Component extends React.PureComponent {
```

Comment:
is this testing the reported issue though? the reporter mentions async functions being involved (which makes them trnaspiled to generators)

isnt this code here having location? (its not inserted by babel)

Reply:
by whatever reason `async/await` after being processed by `regenerator` has `path.node.generator === true`.
at the same time with `env` preset both `async/await` and `function*` are replaced by regenerator and don't have locations [demo](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAEDCD2BbADvAdgUzQFwIzQwA9ssATGAJQzGGwDoAFAVwCcMEV0ttoBvAKGjRIATzTBoAcwzYAImGxgAFAEo-godDbZWaYQHcwASx4MWSIxAx02EeCABuGJWiYgQKjQF8NbNKQwWVXVNaGB0CB5SBTBoAF5oAG06FOwAC0s6aTkY1QBdDSFtXWhoxW9-H35QSBgOVEwcACYCYjJKalpGVnYkBu4Q4QgxCWz5RWCBUOKWPTBDE2gzCysbDDtHZ1d3TyEfIrJAycKwiKiY-KSUunTMsdyVAumZErKwCq8gA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=6.26.0&envVersion=1.6.2)

Reply:
hm, this seems terrible for our use case as preset-env is really popular (as it should be - it's primary use case for babel now)

I'll try look into this as soon as possible to see how babel handle this and if there is anything we can do to overcome this

Reply:
@Andarist updated my comment. to test the case reference the original issue